kclvm/tools/src/LSP/src/goto_def.rs

```diff
@@ -126,8 +127,38 @@ fn goto_def_for_import(
     program: &Program,
 ) -> Option<GotoDefinitionResponse> {
     let pkgpath = &stmt.path;
-    let real_path =
+    let mut real_path =
         Path::new(&program.root).join(pkgpath.replace('.', &std::path::MAIN_SEPARATOR.to_string()));
+    let mut positions = get_pos_from_real_path(&real_path);
+
+    if positions.len() == 0 && !real_path.exists() {
+        real_path = PathBuf::new();
+
+        let pkg_root = fetch_metadata(program.root.clone().into())
```

Comment:
Extract the logic for processing metadata into a smaller function for the convenience of other LSP functions.

Reply:
fixed.